src/logic/small.lean

```diff
@@ -77,23 +76,23 @@ begin
   apply_instance,
 end
 
-@[priority 100]
-instance small_of_fintype (α : Type v) [fintype α] : small.{w} α :=
-begin
-  rw small_congr (fintype.equiv_fin α),
-  apply_instance,
-end
-
 theorem small_of_injective {α : Type*} {β : Type*} [small.{w} β]
   (f : α → β) (hf : function.injective f) : small.{w} α :=
 begin
   rw small_congr (equiv.of_injective f hf),
   apply_instance,
 end
 
-@[priority 100]
-instance small_of_encodable (α : Type v) [encodable α] : small.{w} α :=
-small_of_injective _ (encodable.encode_injective)
+instance small_subsingleton (α : Type v) [subsingleton α] : small.{w} α :=
+begin
+  rcases is_empty_or_nonempty α; resetI,
+  { rw small_congr (equiv.equiv_pempty α), apply small_self, },
+  { rw small_congr equiv.punit_of_nonempty_of_subsingleton,
+    apply small_self, assumption, assumption, },
+end
+
+-- We don't define `small_fintype` or `small_encodable` yet,
+-- to keep imports to `logic` to a minimum.
```

Comment:
```suggestion
/-!
We don't define `small_of_fintype` or `small_of_encodable` in this file,
to keep imports to `logic` to a minimum.
-/
```
So that these names are linked in doc-gen